データベース設計

shokoron opened the pull request:
# What
データベース設計の作成


# Why
データベースを設計するため

ER図
https://i.gyazo.com/5c21c730b83f96c5567b8c76e3f467c9.png

shokoron committed:
データベース設計

div-reviewer reviewed (commented):
shokoron様、学習お疲れさまです。
DB設計の初回レビュー依頼頂きありがとうございました。プルリクエスト確認させていただきました。

テーブル間のアソシエーションまでよく考えられて作成されていますね！素晴らしいです。
修正点をコメントさせて頂きましたので、ご確認をお願い致します。

```ruby
【次回プルリクエスト提出する際の注意点】
①修正が終わりましたら、`コミットとプッシュ`をして、`プルリクエストに修正内容が反映されているか確認`しましょう。
②既に頂いているER図も修正範囲だった場合、次回プルリクエスト提出時には最新の画像の提出をお願い致します。
③googleフォームの送信をお願いいたします。
```

**ご不明点や疑問点ございましたら、チャット・コネクトメンターにご相談ください！
どうぞよろしくおねがいします。**

shokoron committed:
修正しました

shokoron commented on README.md:
```diff
@@ -1,24 +1,73 @@
+| nickname           | string | null: false               |
+| email              | string | null: false, unique: true |
+| encrypted_password | string | null: false               |
+| name               | string | null: false               |
```
修正完了しました。よろしくお願いします！ER図：https://i.gyazo.com/f2fcd86a527572921fc5ffe88f88c2fd.png

shokoron commented on README.md:
```diff
@@ -1,24 +1,73 @@
+| condition          | integer | null: false                    |
+| shipping_charge    | integer | null: false                    |
+| shipping_area      | integer | null: false                    |
+| shipping_days      | integer | null: false                    |
```
修正しました。

shokoron commented on README.md:
```diff
@@ -1,24 +1,73 @@
+| condition          | integer | null: false                    |
+| shipping_charge    | integer | null: false                    |
+| shipping_area      | integer | null: false                    |
+| shipping_days      | integer | null: false                    |
```
修正しました。

shokoron commented on README.md:
```diff
@@ -1,24 +1,73 @@
+| shipping_area      | integer | null: false                    |
+| shipping_days      | integer | null: false                    |
+| price              | integer | null: false                    |
+| user               | integer | null: false, foreign_key: true |
```
修正しました。

shokoron commented on README.md:
```diff
@@ -1,24 +1,73 @@
-* ...
+- belongs_to :user
+- has_one :order
+- has_one :address
```
修正しました。

shokoron commented on README.md:
```diff
@@ -1,24 +1,73 @@
+| Column             | Type    | Options                        |
+| ------------------ | ------  | ------------------------------ |
+| postal_code        | string  | null: false                    |
+| prefecture         | integer | null: false                    |
```
修正しました。

shokoron commented on README.md:
```diff
@@ -1,24 +1,73 @@
+| prefecture         | integer | null: false                    |
+| city               | string  | null: false                    |
+| street_address     | string  | null: false                    |
+| building_name      | string  | null: false                    |
```
修正しました。

shokoron commented on README.md:
```diff
@@ -1,24 +1,73 @@
+### Association
+
+- belongs_to :item
+- belongs_to :order
```
修正しました。

div-reviewer reviewed (commented):
shokoron様、学習お疲れさまです。
DB設計の再レビュー依頼頂きありがとうございました。プルリクエスト確認させていただきました。

完成まで後もう少しですね。最後まで頑張りましょう。
修正点をコメントさせて頂きましたので、ご確認をお願い致します。

```ruby
【次回プルリクエスト提出する際の注意点】
①修正が終わりましたら、`コミットとプッシュ`をして、`プルリクエストに修正内容が反映されているか確認`しましょう。
②既に頂いているER図も修正範囲だった場合、次回プルリクエスト提出時には最新の画像の提出をお願い致します。
③googleフォームの送信をお願いいたします。
```

**ご不明点や疑問点ございましたら、チャット・コネクトメンターにご相談ください！
どうぞよろしくおねがいします。**

shokoron commented on README.md:
```diff
@@ -1,24 +1,75 @@
+
+
+
+## address テーブル
```
修正しました。

shokoron commented on README.md:
```diff
@@ -1,24 +1,75 @@
+
+### Association
+
+- belongs_to :item
```
修正しました。

shokoron commented on README.md:
```diff
@@ -1,24 +1,75 @@
+
+- belongs_to :user
+- belongs_to :item
+
```
修正しました。

shokoron replied:
ER図も添付いたします。https://i.gyazo.com/3e200c4451f7fafb32fcca4f07dc6243.png

shokoron committed:
修正しました

koyamatakumi reviewed (commented):
shokoron様
レビュー依頼フォームの送信、ありがとうございます！

ER図を拝見すると、itemsとaddresses間で線が引かれてしまっている為消しておきましょう。

修正いただき改めてレビュー依頼フォームの送信をお願い致します。

shokoron committed:
修正しました。

ご確認の程よろしくお願いいたします。
ER図；https://i.gyazo.com/164f9477c3b45aac8d00f76ae7b96913.png

ushi10 commented:
レビュー依頼ありがとうございます。READMEならびにER図を確認いたしました。
こちら、LGTMとさせていただきます。実装お疲れさまでした。

改めて記述内容をご確認いただき、問題がない様子でしたらこちらのブランチを忘れずマージしましょう。
マージ後、ローカルのmainブランチにて忘れずにプルをしましょう。
 その後、新しいブランチを作成しissueを確認しながら実装に取り組みましょう。
